pages/words.js

```diff
@@ -0,0 +1,215 @@
+  const [pageIndex, setPageIndex] = useState(0)
+  const loaderRef = useRef(null)
+  const dispatch = useAppDispatch()
+  const chars = useAppSelector(slctr.getChars)
+  const searchType = useAppSelector(slctr.getSearchType)
+  const allWords = useAppSelector(slctr.getWords)
+  const wordLenInterval = useAppSelector(slctr.getLenInterval)
+  const caseInsensitive = useAppSelector(slctr.getCaseInsensitive)
+  const diacriticsInsensitive = useAppSelector(slctr.getDiacriticsInsensitive)
+  const pageSize = 50
+  const [wordsJsx, setWordsJsx] = useState([])
+
+  function wordsToJsx(wordsToShow) {
+    return wordsToShow.map((word, idx) => {
+      return (
+        <Typography key={idx} sx={{ color: 'result.main' }}>
+          {word}
+        </Typography>
+      )
+    })
+  }
+
+  const filterMethodsJsx = Object.entries(searchTypeEnum).map(
+    ([key, value]) => (
+      <MenuItem key={key} value={value}>
+        {value}
+      </MenuItem>
+    )
+  )
+
+  const addPage = useCallback(async () => {
+    if (isLoading) return
+
+    setIsLoading(true)
+
+    const newWords = allWords.slice(
+      pageSize * pageIndex,
+      pageSize * (pageIndex + 1)
+    )
+    setWordsJsx((prevWords) => [...prevWords, wordsToJsx(newWords)])
+    setPageIndex((prevIndex) => prevIndex + 1)
+    setIsLoading(false)
+  }, [isLoading, allWords, pageIndex])
+
+  useEffect(() => {
+    setIsLoading(true)
+    setWordsJsx(wordsToJsx(allWords.slice(0, pageSize)))
+    setPageIndex(1)
+    setIsLoading(false)
+  }, [allWords])
+
+  useEffect(() => {
+    const loaderRefLocal = loaderRef.current
+    const observer = new IntersectionObserver((entries) => {
+      const target = entries[0]
+      if (target.isIntersecting) {
+        addPage()
+      }
+    })
+
+    if (loaderRefLocal) {
+      observer.observe(loaderRefLocal)
+    }
+
+    return () => {
+      if (loaderRefLocal) {
+        observer.unobserve(loaderRefLocal)
+      }
+    }
+  }, [addPage])
+
+  const onCharsChange = useCallback(
+    (value) => {
+      dispatch(setChars({ value }))
+    },
+    [dispatch]
+  )
+
+  const onSearchTypeChange = useCallback(
+    (value) => {
+      dispatch(setSearchType({ value }))
+    },
+    [dispatch]
+  )
+
+  const onLenIntervalChange = useCallback(
+    (value) => {
+      dispatch(setLenInterval({ value }))
+    },
+    [dispatch]
+  )
+
+  const onCaseInsensitiveChange = useCallback(
+    (value) => {
+      dispatch(setCaseInsensitive({ value }))
+    },
+    [dispatch]
+  )
+
+  const onDiacriticsInsensitiveChange = useCallback(
+    (value) => {
+      dispatch(setDiacriticsInsensitive({ value }))
+    },
+    [dispatch]
+  )
+
+  return (
+    <>
+      <Box className={layout_styles.page}>
+        <AppBar />
+        <Box
+          component="main"
+          className={layout_styles.main_decoder}
+          sx={{ color: 'primary.main' }}
+        >
+          <Box className={layout_styles.inputs_box}>
+            <Box>
+              <TextField
```

Comment:
Při zadávání znaků v orientaci na výšku a někdy i na šířku na mobilu (iphone safari/chrome) zmizí prvních několik nalezených slov, což jsou zrovna ta nejdůležitější.

Tady o tom něco píšou: https://stackoverflow.com/a/77815388/5259610
A tady je naznačeno, jak detekovat změnu velikosti pomocí JS, asi i na safari: https://stackoverflow.com/a/65092685/5259610
V návaznosti na to by asi bylo potřeba upravit CSS, aby si poradilo i se zobrazením na obrazovce zmenšené o klávesnici.